teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -187,6 +189,20 @@ interface TestRunsLocator {
     fun forProject(projectId: ProjectId): TestRunsLocator
     fun withStatus(testStatus: TestStatus): TestRunsLocator
     fun all(): Sequence<TestRun>
+    fun all(fields: EnumSet<TestRunFields>): Sequence<TestRun>
+}
+
+enum class TestRunFields {
```

Comment:
Are you sure we really need to allow excluding every possible field here? You said that "The details of the test run can be extremely large" so maybe it's enough to skip `details` and `ignoreDetails` fields? In that case it'll be enough to add a function `withoutDetails` to `TestRunsLocator`, this will greatly simplify API and implementation. Also note that currently if someone exclude `Build`, `Name` or `Test` field, and then try to access corresponding property in `TestRun`, he'll get an exception, it may be unexpected.

Note that providing access to all possible options in TeamCity REST API is not a goal for this library. It should provide convenient API for real use cases.